Submitted models/marble husky sensor config 3

bfotheri opened the pull request:
The marble virtual track team has updated their systems vehicles since we submitted their models. We talked with Angela and she said if they had built the vehicles perhaps you guys would consider these additional sensor configurations. Here are the images showing the two vehicles and reflecting this vehicle sensor configuration. 


![tmp_1596226782657](https://user-images.githubusercontent.com/25831722/89076816-3ba00f80-d33e-11ea-95cb-b3e152b5477f.jpg)
![tmp_1596226820604](https://user-images.githubusercontent.com/25831722/89076835-3fcc2d00-d33e-11ea-8b20-b6e71828e358.jpg)

shakeebbb commented:
![version 3](https://user-images.githubusercontent.com/36937615/89111424-cd794c80-d412-11ea-8d98-0238e9825a10.jpg)
Here is the photo of the UAV with the sensor suite that is on the simulated model except additional side cameras for Artifact Detection. We used this UAV at the Urban Circuit event and we are planning to add additional cameras for the Artifact Detection.

mjcarroll commented on submitted_models/marble_husky_sensor_config_3/.vscode/c_cpp_properties.json:
```diff
@@ -0,0 +1,20 @@
+{
```
Please remove editor support files `.vscode/*`

bfotheri replied:
A commit removing this on my fork has been made.

mjcarroll commented:
![image](https://user-images.githubusercontent.com/279701/89576481-f6c02100-d7f4-11ea-9439-64362a1ccf32.png)

mjcarroll commented:
Sensor data and coordinate frames look good
![image](https://user-images.githubusercontent.com/279701/89576896-8534a280-d7f5-11ea-8427-31676b2d833a.png)

mjcarroll commented on submitted_models/marble_husky_sensor_config_3/meshes/VLP16_base_2.dae:
```diff
@@ -0,0 +1,89 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no" ?>
```
Can this be deduplicated with VLP base 1?

bfotheri replied:
There is a small little knob representing the plugin of the sensor that differs base 1 and base 2 but the design could be simplified and the knobless base could be used for the top and bottom. Let me know if that's what we should do.

mjcarroll commented on submitted_models/marble_husky_sensor_config_3/package.xml:
```diff
@@ -0,0 +1,28 @@
+  <name>marble_husky_sensor_config_3</name>
+  <version>0.1.0</version>
+  <description>The marble_husky_sensor_config_3 package
+    - Has two rgbd camera Realsense D435
```
There is only one RGBD camera in the SDF file

bfotheri replied:
You are correct. The package.xml has been updated to more accurately reflect the vehicle configuration.

mjcarroll commented on submitted_models/marble_husky_sensor_config_3/specifications.md:
```diff
@@ -0,0 +1,86 @@
+The following specific sensors are declared payloads of this vehicle:
+
+* Trossen ScorpionX MX-64 Robot Turret, modeled by JointStateController and JointStatePublisher plugins.
+* D435i RGBD Camera (x3), modeled by rgbd_camera plugin
```
There is only one RGBD camera int he sdf file, is this correct?

bfotheri replied:
This is correct. The specifications.md file has been updated to more accurately reflect the vehicle config.

mjcarroll commented on submitted_models/marble_husky_sensor_config_3/specifications.md:
```diff
@@ -0,0 +1,86 @@
+## Description
+This configuration is based on Clearpath Robotics Husky ground robot. The marble sensor suite is located at the front of the husky and includes stationary sensors and some sensors and a light on a pan/tilt mechanism.  
+
+## Usage Instructions
+The robot can be used in the same manner as the COSTAR Husky robot.  It accepts twist messages on the cmd_vel topic.  The gimbal can be controlled using the test_gimbal.sh script (in this folder).  There are also scripts to print the current angles from the pan and tilt axes of the gimbal (pan_echo.sh and tilt_echo.sh).
+
+The gimbal can be controlled via the following topics:
+pan_tilt/pan_rate_cmd_double (send a std_msgs/Double message)
+pan_tilt/tilt_rate_cmd_double (send a std_msgs/Double message)
+
+The position of the pan and tilt axes can be accessed through the joint_state topic.  The name of each axis is available inside that message, but as of today the pan and tilt can be accessed as follows:
+Pan: joint_state/position[4]
+Tilt: joint_state/position[5]
```
Is this variant supposed to have the gimbal?  I don't see any evidence of it.  It should either be added to the SDF or removed from the documentation for this model.

bfotheri replied:
It does not have it a gimbal. The specifications.md file has been updated to more accurately reflect the vehicle config.

mjcarroll commented on submitted_models/marble_husky_sensor_config_3/specifications.md:
```diff
@@ -0,0 +1,86 @@
+### Diversions from Physical Hardware of MARBLE Husky
+Computers were installed in the payload area of the husky and these have been roughly modeled (the rail and computer bay are shown in the model.sdf model).  The MARBLE vehicle uses an AMD Ryzen processor (32-core) with 64 GB of RAM.  It has a cooling system as well.  
+
+## X4 Validation and Specification Links
```
X4 should be marble?

bfotheri replied:
This has been changed to marble husky. Thanks for the catch!

bfotheri commented:
I'm also adding some forward and rear facing lights that are on the vehicle but went unmodeled.

bfotheri commented on submitted_models/marble_husky_sensor_config_3/model.sdf:
```diff
@@ -0,0 +1,1015 @@
+        <camera name="camera_down">
+            <horizontal_fov>1.5184</horizontal_fov>
+            <lens>
+                <intrinsics>
+                    <fx>277.1</fx>
+                    <fy>277.1</fy>
+                    <cx>640.5</cx>
+                    <cy>360.5</cy>
+                    <s>0</s>
+                </intrinsics>
```
Sensor configuration 1 was correct. I will revert that.

bfotheri commented on submitted_models/marble_husky_sensor_config_3/model.sdf:
```diff
@@ -0,0 +1,1015 @@
+      <!-- Based on RPLidar S1 -->
+      <sensor name="planar_laser" type="gpu_ray">
+          <pose>0.485 0 0.166 0 -0 0</pose>
+          <update_rate>20</update_rate>
```
Also a mistake, will revert that. Potentially a mix up with the 3D lidar update rate.

mjcarroll commented on submitted_models/marble_husky_sensor_config_3/specifications.md:
```diff
@@ -0,0 +1,72 @@
+* Vehicle Links:
+  * https://clearpathrobotics.com/husky-unmanned-ground-vehicle-robot/
+
+* Sensor specification links:
+  * D435i RGBD Camera - https://www.intelrealsense.com/depth-camera-d435i/
+  * Ouster 3D Lidar (64 Channel) - https://ouster.com/products/os1-lidar-sensor/
+  * RPLidar S1 Planar Lidar - https://www.slamtec.com/en/Lidar/S1Spec 
+  * IMU: Microstrain 3DM-GX5-25 - datasheet: https://www.microstrain.com/sites/default/files/applications/files/3dm-gx5-25_datasheet_8400-0093_rev_n.pdf
+    * Explanation of sensor parameter derivations:
+	We derived the stddev terms as follows:
+
+	accelerometer noise density = 0.00002 g/sqrt(Hz)
+		=> convert to m/s^2 => 1.962e-4 m/s^2
+	gyro noise density = 0.005 deg/s/sqrt(Hz)
+		=> convert to rad/sec => 8.72664e-5 radians
+
+	Other terms are difficult to extract from datasheet, so we used similar terms to previous IMU models proposed (of similar or worse quality) such as the ADIS 16448 (which has worse performance than this IMU).
```
Friendly ping @bfotheri

bfotheri replied:
Hey! @mjcarroll I just added the cameras reference. I haven't fixed the floating sensors yet however. I recently received a STEP file of the whole robot and am planning to convert parts to .dae files then add them into the visualization to eliminate the floating sensors.

acschang reviewed (commented):
This sensor configuration will be considered for the final circuit.

- Please connect floating sensors to the existing mesh.
- The provided documentation for the HD MIPI cameras lists the cameras to have a 808x608 resolution, 50 Hz frame rate, and the listed lens has a 80 degree FoV.  The parameters listed for the modeled sensor are 1280x960 resolution, 15 Hz frame rate, and a 60 degree FoV.  Please either correct the discrepancy or provide an explanation for the simulated sensor parameters differing from the provided documentation.
- The validation data is missing. Please complete the validation testing as required by the [Simulation Model Preparation Guide](https://subtchallenge.com/resources/Simulation_Model_Preparation_Guide.pdf) and submit the required data.

nkoenig commented:
The bounding box for this model is
```
Min[-0.492701 -0.34339 -0.144193] Max[0.5479 0.343206 0.68003]
```

mjcarroll commented:
@bfotheri Friendly ping, there are some requested changes from @acschang

bfotheri commented:
Hey guys, sorry for the delay. So Marble has updated their vehicle since this was model was done ~9 month ago and I have since used this sensor configuration as a base model, added the additional cameras, changed out 2 D435i's for 2 picco flexx cameras. 

Would you prefer me to 1) Update this sensor configuration to reflect the most current one or 2) Create a new pull request for the new sensor configuration?

bfotheri commented:
Also @acschang 

For 1) & 2) I'll push a commit to address that. For 3) I believe MARBLE has collected some of this data at least. I'll discuss the required tests with them and ask when we can have this done by.
